MovieQuiz/Services/NetworkClient.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  NetworkClient.swift
+//  MovieQuiz
+//
+//  Created by Aleksandr Eliseev on 11.11.2022.
+//
+
+import Foundation
+
+struct NetworkClient {
+    
+    private enum NetworkError: Error {
+        case codeError
+    }
+    
+    
+    func fetch(url: URL, handler: @escaping (Result<Data, Error>) -> Void) {
+        let request = URLRequest(url: url)
+        
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+            // Check if an error occured
+            if let error = error {
+                handler(.failure(error))
+                return
+            }
+            
+            // Check if a successful response was received
+            if let response = response as? HTTPURLResponse,
+                response.statusCode < 200 || response.statusCode >= 300 {
+                    handler(.failure(NetworkError.codeError))
+                    return
+                }
+                
+                // return the data
+                guard let data = data else { return }
+                handler(.success(data))
+            }
+            task.resume()
```

Comment:
Не забывайте использовать `control+i`
```suggestion
        }
        task.resume()
```